src/toil/jobStores/awsJobStore.py

```diff
@@ -620,36 +652,44 @@ def _registerFile(self, jobStoreFileID,
         :param jobStoreID: optional ID of the job owning the file, only allowed for first version of
                            file
 
-        :param newVersion: the file's new version or None if the file is to be registered without
+        :param newFile: the file's new version or None if the file is to be registered without
                            content, in which case jobStoreId must be passed
 
-        :param oldVersion: the expected previous version of the file or None if newVersion is the
+        :param oldVersion: the expected previous version of the file or None if newFile is the
                            first version or file is registered without content
         """
-        # Must pass newVersion if passing oldVersion
-        assert oldVersion is None or newVersion is not None
-        attributes = dict(bucketName=bucketName)
-        if newVersion is not None:
-            attributes['version'] = newVersion
-        if jobStoreID is not None:
-            attributes['jobStoreID'] = jobStoreID
+        attributes = {}
```

Comment:
I feel like this whole block that determines the attributes dict should be a method in FileInfo.